Cubical/HITs/FreeGroup/Base.agda

```diff
@@ -0,0 +1,28 @@
+{-
+
+This file contains:
+
+- An implementation of the free group of a type of generators as a HIT
+
+-}
+{-# OPTIONS --safe #-}
+
+module Cubical.HITs.FreeGroup.Base where
+
+open import Cubical.Foundations.Prelude
+
+private
+  variable
+    ℓ : Level
+
+data FreeGroup (A : Type ℓ) : Type ℓ where
+  η     : A → FreeGroup A
+  m     : FreeGroup A → FreeGroup A → FreeGroup A
+  e     : FreeGroup A
+  inv   : FreeGroup A → FreeGroup A
+  assoc : ∀ x y z → m x (m y z) ≡ m (m x y) z
+  idr   : ∀ x → x ≡ m x e
+  idl   : ∀ x → x ≡  m e x
+  invr  : ∀ x → m x (inv x) ≡ e
+  invl  : ∀ x → m (inv x) x ≡ e
+  trunc : ∀ x y → ∀ (p q : x ≡ y) → p ≡ q
```

Comment:
It is better to use `isSet (FreeGroup A` instead of `∀ x y → ∀ (p q : x ≡ y) → p ≡ q`

Reply:
Same as above.

Reply:
I think using `isSet` is clearer so let's go for it